### PR TITLE
templates: Create GitHub nomination template

### PR DESCRIPTION
**Pull request by keith-zephyr**
Create a template for nominating GitHub users for an elevated
permission levels.

Signed-off-by: Keith Short <keithshort@google.com>

### CODE REVIEW

**Review by anasnashif — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
>  ** [anasnashif](/anasnashif) ** approved these changes [ yesterday ](#pullrequestreview-749371626)

@nashif care to try again?

File: .github/ISSUE_TEMPLATE/contributor-nomination.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+---
+
+# Background
+
+The [TSC Project Roles] defines the main roles for the Zephyr Project, including
+Maintainter, Collaborator, and Contributor.
+
+TODO: define new roles for Contributors with extra rights? This would correspond
+to the Github "Triage" permission level. This template uses Contributor+ as a
+placeholder.
```
Comment (mbolivar-nordic) [inline .github/ISSUE_TEMPLATE/contributor-nomination.md:17]: Process WG:

- keep Contributor level; do not introduce Contributor+ by that name or any other
- add a line to the documentation for Contributor which describes the process for opening an issue that fills out this template

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Thanks for taking this on; I've recorded the next steps above.

**Comment by keith-zephyr** [comment]
> > ** [anasnashif](/anasnashif) ** approved these changes [ yesterday ](#pullrequestreview-749371626)
> 
> @nashif care to try again?

I believe this was Anas testing out some permission levels and doesn't represent an actual vote.

**Comment by mbolivar-nordic** [comment]
Process WG: wait for update and continue review here

**Comment by mbolivar-nordic** [comment]
Process WG follow up actions:

- reassign roles within the project settings
- send announcement about the change (which will imply loss of privileges in some cases)

**Comment by nashif** [comment]
@keith-packard  can you fix the issues and move from draft?

**Comment by keith-zephyr** [comment]
> @keith-packard can you fix the issues and move from draft?

I was OOO the end of last week.  I'll get these issues cleaned up by tomorrow (Sept. 27)

**Review by nashif — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/nomination.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+title: ''
+labels: Role Nomination
+assignees: ''
+
+---
+
+# Background
+
+The [TSC Project Roles] defines the main roles for the Zephyr Project, including
+Maintainter, Collaborator, and Contributor.
```
Comment (mbolivar-nordic) [inline .github/ISSUE_TEMPLATE/nomination.md:13]:
```suggestion
Maintainer, Collaborator, and Contributor.
```

File: doc/development_process/project_roles.rst [hunk 3]
```diff
@@ -49,6 +48,26 @@ Contributors are granted the following rights and responsibilities:
 * Responsibility to follow the project’s code of conduct
   (https://github.com/zephyrproject-rtos/zephyr/blob/main/CODE_OF_CONDUCT.md)
 
+Contributors are initially only given `Read
+<https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-permission-levels-for-an-organization>`_
+access to the Zephyr GitHub repository. Specifically, at the Read access level,
+Contributors are not allowed to assign reviewers to their own pull-requests. An
```
Comment (mbolivar-nordic) [inline doc/development_process/project_roles.rst:54]: I think we generally us "pull request" without a hyphen. (Applies elsewhere in the PR.)

File: doc/development_process/project_roles.rst [hunk 4]
```diff
@@ -210,13 +240,16 @@ Roles / Permissions
...
 MAINTAINERS File
 ################
 
 Generic guidelines for deciding and filling in the Maintainers' list
 
-* The MAINTAINERS file shall replace the CODEOWNERS file and will be used
-  for both setting assignees and reviewers.
+* The :zephyr_file:`MAINTAINERS.yaml` file shall replace the
+  :zephyr_file:`CODEOWNERS` file and will be used for both setting assignees and
+  reviewers.
```
Comment (mbolivar-nordic) [inline doc/development_process/project_roles.rst:252]: @nashif do we really want to merge this in this PR? It seems premature, since #38725 is still open.

**Review by mbolivar-nordic — COMMENTED** [review]
Thanks! A couple of nitpicks and one question.

File: doc/development_process/project_roles.rst (hunk 4 above)
Comment (keith-zephyr) [reply doc/development_process/project_roles.rst:252]: Note that I'm only adding file links to the existing text here.

Comment (mbolivar-nordic) [reply doc/development_process/project_roles.rst:252]: resolved offline

**Comment by mbolivar-nordic** [comment]
Process WG:

- Get agreement on this PR
- @nashif: should bring it up at the TSC, @mbolivar-nordic to discuss the changes to permissions 
- Escalated to TSC after that point

File: doc/development_process/project_roles.rst (hunk 3 above)
Comment (keith-zephyr) [reply doc/development_process/project_roles.rst]: There was existing uses of "pull-request" in this file, but "pull request" (no hyphen) seems to be more widely used so I fixed all references in this file.

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
Added DNM and TSC per discussion in the process WG (see above).

File: doc/development_process/project_roles.rst [hunk 5]
```diff
@@ -49,6 +48,26 @@ Contributors are granted the following rights and responsibilities:
 * Responsibility to follow the project’s code of conduct
   (https://github.com/zephyrproject-rtos/zephyr/blob/main/CODE_OF_CONDUCT.md)
 
+Contributors are initially only given `Read
+<https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-permission-levels-for-an-organization>`_
+access to the Zephyr GitHub repository. Specifically, at the Read access level,
+Contributors are not allowed to assign reviewers to their own pull requests. An
+automated process will assign reviewers, or you may share the pull request on
+the `Zephyr devel mailing list <https://lists.zephyrproject.org/g/devel>`_.
```
Comment (mbolivar-nordic) [inline doc/development_process/project_roles.rst:56]: Another thing people do is ask for reviews on Discord, if you want to add a link to that here as well. Not a blocking comment, just something that occurred to me.

File: .github/ISSUE_TEMPLATE/nomination.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+Provide the following information about the GitHub user:
+
+1. Full Name
+1. GitHub username
+1. Organization (optional)
+
+## Supporting Documents
+
+Add links to 3-5 GitHub pull requests, in the Zephyr project, authored or
+reviewed by the GithHub user that demonstrate the user's dedication to the
```
Comment (pdgendt) [inline .github/ISSUE_TEMPLATE/nomination.md:36]: Typo
```suggestion
reviewed by the GitHub user that demonstrate the user's dedication to the
```

**Pull request merged**